contrib/babelfishpg_tds/src/backend/tds/tdsprotocol.c

```diff
@@ -290,7 +294,16 @@ GetTDSRequest(bool *resetProtocol)
 			resetCon->messageType = messageType;
 			resetCon->status = (status & ~TDS_PACKET_HEADER_STATUS_RESETCON);
 
+			/*
+			 * Set resetTdsConnectionFlag to true so that we avoid
+			 * sending any env change token for the USE DB command
+			 * which will get executed.
+			 */
+			resetTdsConnectionFlag = true;
+			TdsSetDbContext();
+			resetTdsConnectionFlag = false;
```

Comment:
Is there a possibility of `resetTdsConnectionFlag` value getting leaked in case there is an error in `TdsSetDbContext()`? If yes then a TRY/CATCH here will help.

Reply:
There is no possibility, if you checked TdsSetDbContext function it would be apparent that there is a TRY/CATCH inside already and in the CATCH we throw FATAL error which disconnects the sessions